src/kubelet/metric/transform.go

```diff
@@ -0,0 +1,54 @@
+package metric
+
+import (
+	"fmt"
+
+	"github.com/newrelic/nri-kubernetes/v2/src/definition"
+)
+
+// OneMetricPerLabel transforms a map of labels to FetchedValues type,
+// which will be converted later to one metric per label.
+// It also prefix the labels with 'label.'
+func OneMetricPerLabel(rawLabels definition.FetchedValue) (definition.FetchedValue, error) {
+	return PrefixFromMapString("label.")(rawLabels)
+}
+
+// PrefixFromMapString gets a value assumed to be a map and converts it to a set of values given a prefix, which should
+// includethe separating dot (`.`).
+// For example, calling PrefixFromMapString("prfx.") on a map containing {"foo": "bar", "boo": "baz"} will cause
+// { "prfx.foo": "bar", "prfx.boo": "baz"}
+// to be added to the group.
+func PrefixFromMapString(prefix string) func(mapValue definition.FetchedValue) (definition.FetchedValue, error) {
```

Comment:
Can't this be casted to `map[string]interface{}` maybe, so the function can be de-duplicated with `PrefixFromMapInt`? Or maybe `switch` case could be used somehow?